.github/workflows/static-analysis.yml

```diff
@@ -0,0 +1,113 @@
+name: Luau static file analysis
```

Comment:
Can we integrate this into benchmark.yml instead? :)
We are likely going to add more categories than just the static analysis, eg compiler, JIT, in the future, and per previous conversations it's going to be easier for us to manage a single action.